sunpy/lightcurve/sources/stereo.py

```diff
@@ -0,0 +1,906 @@
+			else:
+				out[name] = column.astype(np.object).filled(np.nan)
+		else:
+			out[name] = column
+
+		if out[name].dtype.byteorder not in ('=', '|'):
+			out[name] = out[name].byteswap().newbyteorder()
+
+	return DataFrame(out)
+
+
+
+class LETLightCurve(LightCurve):
+
+	"""
+	STEREO LET LightCurve. Provides data from as back as 2006. Currently the LightCurve supports only Single File Load !!
+
+	Parameters
+	----------
+	timerange: sunpy.time.TimeRange
+	   time range for which data is to be downloaded.
+	   Default value -  TimeRange('2007-01-01','2008-06-01')   
+
+	duration_of_average: string
+	   Default value - 10*u.min
+	   Possible values - 1*u.min, 10*u.min, 1*u.h, 1*u.d, 27*u.d   
+	   #corresponding to duration over which data is averaged
+
+	type_of_data:  string
+	   Possible values - depends on other parameters
+	   if duration_of_average = 27*u.d:
+	   #Possible Values: summed, narrow
+	   else:
+	   #Possible values: sectored, standard, summed
+
+	specie:  string
+	   Possible values - depends on other parameters
+	   if type_of_data = 'Sectored' and duration_of_average in [ 1*u.min, 10*u.min, 1*u.h, 1*u.d]:
+	   #Possible values: CNO_hi,CNO_lo, Fe_hi, Fe_lo, H_lo, He3_lo, He4_hi, He4_lo, He_lo, NeMgSi_hi, NeMgSi_lo
+	   else:
+	   #Possible values: Al, Ar, C, Ca, Fe, H, He, He3, He4, Mg, N, Na, Ne, Ni, O, S, Si
+
+	stereo_spacecraft: string   
+	   Default value - ahead
+	   Possible values - ahead, behind    
+	   # corresponding to spacecraft location
+
+	Examples
+	--------
+	>>> import os
+	>>> import sunpy.data.test
+	>>> filepath = sunpy.data.test.rootdir
+	>>> from sunpy import lightcurve as lc
+	>>> let = lc.LETLightCurve._parse_txt(os.path.join(filepath , 'let/Ar_ahead_2006_318_level1_11.txt'))
+	>>> let = lc.LETLightCurve(let[1],let[0])
+	>>> let.peek()
+
+	References
+	----------
+	| http://www.srl.caltech.edu/STEREO/Level1/LET_public.html
+
+	"""
+
+	def peek(title="LET electron/proton Flux"):
+		"""Plots LET light curve in the usual manner"""
+		figure = plt.figure()
+		ax = plt.gca()
+
+		header, data = _parse_txt('Ar_ahead_2006_318_level1_11.txt')
+		dates = matplotlib.dates.date2num(data['Datetime'].astype(datetime))
+
+		colors = ['Green','Red', 'Blue','SeaGreen','Tomato','SlateBlue','Orange',
+							'Purple','Magenta','Chocolate','MediumVioletRed', 'Teal','Navy','Indigo']
+
+		figure.delaxes(ax)
+		axes = figure.add_axes([0.1, 0.15, 0.55, 0.8])
+
+		if header[1][:4] == 'Flux':
+			#27 day data
+			num_energy_bins = (len(header)-1)/2
+			for i,line in enumerate(header):
+				if i >= 1 and i < num_energy_bins:
+					axes.plot_date(dates, data['col' +str(3+i)].ffill(), '-', label= line[9:], color=colors[i], lw=0.5)
+			axes.set_yscale("log",nonposy='mask')
+			
+		elif header[1][:22] == 'Column 6: LET Livetime' and header[2][9] == '-':
+			#Sectored data
+			num_energy_bins = header[2][header[2].index(':')+2:header[2].index('s')-1]
+			data = data.replace(-9999.9,float('nan'))
```

Comment:
wouldn't be better to do these data conversions during reading? as it is now they will just be converted when the user executes `peek` and not `plot` directly.  
This will produce different "results" between
   -  reads it, peek it and then plot it  and,
   -  reads it, and plot it.